groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -3255,23 +3514,30 @@ Variant_Base<HEAD, TAIL...>::Variant_Base(const Variant& original)
     if (!original.valueless_by_exception()) {
         BloombergLP::bslstl::Variant_CopyConstructVisitor<Variant_Base>
             copyConstructor(this);
-        BSLSTL_VARIANT_VISITID(void, Variant, copyConstructor, original);
+        BSLSTL_VARIANT_VISITID(void, copyConstructor, original);
+            // in C++03, if there are nonunique alternatives,
+            // 'BSLSTL_VARIANT_VISITID' will always use the first nonunique
+            // alternative, which may result in the wrong 'd_type' being set.
+        d_type = original.d_type;
```

Comment:
This is already only C++03 code (or rather, the code that is not full C++11 implementation)

Reply:
Ah, you're right. I think it was the word "may" that confused me, as I interpreted it as meaning "if this is C++03" rather than "if the currently active alternative is not the first one with its type". (I'm not going to insist that you change the comment, though. I should have looked more carefully in the first place.)